editor/src/components/inspector/inspector.tsx

```diff
@@ -448,26 +448,36 @@ const DefaultStyleTargets: Array<CSSTarget> = [cssTarget(['style'], 0), cssTarge
 export const InspectorEntryPoint: React.FunctionComponent = betterReactMemo(
   'InspectorEntryPoint',
   () => {
-    const { selectedViews, jsxMetadataKILLME } = useEditorState((store) => {
-      return {
-        selectedViews: store.editor.selectedViews,
-        jsxMetadataKILLME: store.editor.jsxMetadataKILLME,
-      }
-    }, 'InspectorEntryPoint')
+    const selectedViews = useEditorState(
+      (store) => store.editor.selectedViews,
+      'InspectorEntryPoint selectedViews',
+    )
+    const rootViewsForScene: Array<TemplatePath> = useEditorState(
+      (store) => {
+        const possibleRootComponent = store.editor.jsxMetadataKILLME.find((m) =>
+          TP.pathsEqual(m.scenePath, selectedViews[0]),
+        )
+        if (possibleRootComponent != null) {
+          return possibleRootComponent.rootElements.map((e) => e.templatePath)
+        } else {
+          return []
+        }
+      },
+      'InspectorEntryPoint',
+      (oldTemplatePaths, newTemplatePaths) => {
```

Comment:
hmm, maybe I can ditch this now that @Rheeseyb  's TemplatePath pr is merged?

Reply:
It would still create a new array, so probably worth including a variant of this, but it could be simplified to use reference equality for each element in the array.

Reply:
ah, actually TP.pathsEqual starts by checking === equality

Reply:
We have the `arrayEquals` function, better to use that here.

Reply:
thanks! I fixed it on master in 79594140